src/Symfony/Component/PropertyInfo/Util/PhpStanTypeHelper.php

```diff
@@ -161,6 +166,8 @@ private function extractTypes(TypeNode $node, NameScope $nameScope): array
                     return [new Type(Type::BUILTIN_TYPE_INT)];
                 case 'list':
                     return [new Type(Type::BUILTIN_TYPE_ARRAY, false, null, true, new Type(Type::BUILTIN_TYPE_INT))];
+                case 'non-empty-array':
```

Comment:
I'm just wondering: why do we want to add this to 5.3 while the other types are added on 6.1 (see #44451)?

Reply:
I think roo of the problem is fallback case here on L182. If PhpStanTypeHelper doesn't recognize the type, it returns object. This is what causes issues during 5.3 -> 5.4 upgrade for my second use case. Potentially we could change that to return empty array instead, but that's very aggressive change compared to just adding support for non-empty-array (it might break various cases of `@return Foo`, `@return Foo<bar>` and so on. Alternatively, in my opinion backporting changes from #44451 to 5.4 would be fine too. All of those cases that were added there were most likely working fine in 5.3, but got broken in 5.4. `non-empty-array` just happens to be more often used than the rest of them and that's what blowed up in my projects.

Reply:
Thanks, that's exactly why I posted https://github.com/symfony/symfony/pull/44451#pullrequestreview-833806383

@derrabus WDYT?

Reply:
I wouldn't add support for that type here. If we get a proper fallback if the PHPStan extractor reported `null` here, that's the way to go imho.

Reply:
Well then any idea how to properly make a fallback here? We need a mechanism how to detect object, instead of falling back to object on everything that's not recognized.

Reply:
Sorry, your two answers were not displayed when I posted mine which obviously does not make sense in that context. 🙈 

I agree that in that particular case, active support for `non-empty-array` (and possibly `non-empty-list` as well) would be the least invasive fix.

Reply:
I've added non-empty-list :)